src/main/java/com/gmail/yuramitryahin/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package com.gmail.yuramitryahin.service;
+
+import com.gmail.yuramitryahin.dao.OrderDao;
+import com.gmail.yuramitryahin.lib.Inject;
+import com.gmail.yuramitryahin.lib.Service;
+import com.gmail.yuramitryahin.model.Order;
+import com.gmail.yuramitryahin.model.ShoppingCart;
+import com.gmail.yuramitryahin.model.User;
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order order = new Order();
+        order.setUser(shoppingCart.getUser());
+        order.setTickets(shoppingCart.getTickets());
```

Comment:
`order.setTickets(new ArrayList<>(shoppingCart.getTickets()));`
because hibernate delete relations with shoppingCart so you must create clone of your tickets

Reply:
let's fix this. We should explicitly clear the shoppingCart using clear() method from service

Reply:
fixed